Month2/week1/Month2-Project Week 1-Remembering the Bases/VideoGame/Steam.cpp

```diff
@@ -0,0 +1,317 @@
+			std::cin >> ChoiceSelection;
+		}
+		if (ChoiceSelection = 1)
+		{
+			std::cout << "Please enter Your Category Title:" << std::endl;
+			std::string NewCategoryTitle;
+			std::getline(std::cin, NewCategoryTitle);
+			while ( !std::cin.good() || std::cin.peek() == '\n') 
+			{
+				std::cin.clear();
+				std::cin.ignore(1000,'\n');
+				std::cout << "Invalid input, please enter Your Category title or name again:" << std::endl;
+				std::getline(std::cin, NewCategoryTitle);
+
+			}
+			std::cin.ignore(1000, '\n');
+			FCategory NewCategory;
+			NewCategory.SetCategoryTitle(NewCategoryTitle);
+			NewCategory.AddGame(NewGame);
+			CategoryContainer.AddCategory(NewCategory);
+			std::cout << "Game added done" << std::endl;
+			std::cout << "What do you want to do now?" << std::endl;
+			std::cout << "1. Add other Game" << std::endl;
+			std::cout << "2.Back to the Main menu" << std::endl;
+			std::cin >> Choice;
+		}
+		else if (ChoiceSelection = 2)
+		{
+			std::cout << "You want Add the Game to a category?" << std::endl;
+			std::cout << "1.Yes" << std::endl;
+			std::cout << "2.Not" << std::endl;
+			int ChoiseOption;
+			std::cin >> ChoiseOption;
+			while (!std::cin.good() || ChoiseOption > 2 || ChoiseOption < 1) 
+			{
+				std::cout << "Invalid input, Enter your a new choice:" << std::endl;
+				std::cin >> ChoiseOption;
+			}
+			if(ChoiseOption==1){
+			std::cout << "You can see all the Categories in here, Please select one to add the game" << std::endl;
+			for (int CategoryIndex = 0; CategoryIndex < 5; CategoryIndex++) 
+			{
+				std::cout << CategoryIndex << " " << CategoryContainer.GetCategory(CategoryIndex).GetCategoryTitle() << std::endl;
+			}
+			
+			int CategoryChoice;
+			std::cin >> CategoryChoice;
+			while (!std::cin.good() || CategoryChoice < 0 || CategoryChoice >= 5) 
+			{
+				std::cin.clear();
+				std::cin.ignore(1000, '\n');
+				std::cout << "Invalid input, Please repeat Your choice" << std::endl;
+				std::cin >> CategoryChoice;
+			}
+			if(CategoryContainer.GetCategory(CategoryChoice).GetCategoryTitle() == "" || CategoryContainer.GetCategory(CategoryChoice).GetCategoryTitle() == " ")
+			{
+				std::cin.clear();
+				std::cin.ignore(1000, '\n');
+				std::cout << "Sorry the category Does not exist or is not  valid, please make a other choice" << std::endl;
+				std::cin >> CategoryChoice;
+			}
+			CategoryContainer.AddGameToCategory(CategoryChoice,NewGame);
+			std::cout << "GameAdded" << std::endl;
+			std::cout << "1. Add other Game" << std::endl;
+			std::cout << "2. Back to the Main menu" << std::endl;
+			std::cin >> Choice;
+			}
+			else if (ChoiseOption = 2) 
+			{
+				Uncategorized.AddGame(NewGame);
+				std::cout << "GameAdded" << std::endl;
+				std::cout << "Please select a option:" << std::endl;
+				std::cout << "1. Add other Game" << std::endl;
+				std::cout << "2. Back to the Main menu" << std::endl;
+				std::cin >> Choice;
+			}
+		}
+	} while (Choice = 1);
```

Comment:
This is WAY too many lines to have inside a function, even more to have in just one single do-while statement.

Reply:
You need to separate this into like 5 different methods